Assets/Scripts/Projectile.cs

```diff
@@ -0,0 +1,91 @@
+using UnityEngine;
+
+[RequireComponent(typeof(Hit))]
+[RequireComponent(typeof(Rigidbody2D))]
+public class Projectile : MonoBehaviour
+{
+    [SerializeField] private ProjectileData data;
+    [SerializeField] private bool debugMode = false;
+
+    private Rigidbody2D rb;
+    private Hit hitComponent;
+    private float lifeTimer;
+
+    private void Awake()
+    {
+        rb = GetComponent<Rigidbody2D>();
+        hitComponent = GetComponent<Hit>();
```

Comment:
Can we add a `Debug.Error` if the hitComponent isn't found? Nicer than null reference errors - or does `RequireComponent` handle that for you? Rarely used that decorator